app/models/membership.rb

```diff
@@ -0,0 +1,10 @@
+class Membership < ActiveRecord::Base
+  belongs_to :organization, class_name: "Organization"
+
+  # Publisher model will be responsible for casting to the right sub-role
+  # E.g. casting from Publisher to Partner
```

Comment:
I'm a little uncertain about the architecture of how we're going to cast from a Publisher to a Partner, do you have any ideas on the matter?

Reply:
https://api.rubyonrails.org/classes/ActiveRecord/Persistence.html#method-i-becomes